### PR TITLE
make copr_messaging optional

### PR DESCRIPTION
**Pull request by pkking**
#2589 My thoughts stay straightforward, just make the copr_messaging and fedmsg into weak dependency...

### CODE REVIEW

**Comment by praiskup** [comment]
@jmacku, any idea if we could avoid the initial `You have successfully added a new vcs-diff-lint configuration ...` message?  That message spams every single PR.

File: backend/copr_backend/msgbus.py [hunk 1]
```diff
@@ -33,6 +39,8 @@ def message_from_worker_job(style, topic, job, who, ip, pid):
     once we can exepect that people are using copr-messaging module (and thus
     we can change message format without affecting the API).
     """
+    if not copr_messaging_enabled:
+        return None
```
Comment (praiskup) [inline backend/copr_backend/msgbus.py:43]: It looks like this is not needed, what is the traceback you face if you uninstall python-copr-messaging?

MessageSender.announce() calls `bus.announce_job()` only if some bus is configured (which shouldn't be your case) therefore neither `message_from_worker_job` shouldn't be called.

Comment (pkking) [reply backend/copr_backend/msgbus.py:43]: Yes, in today's codebase, there's no need to add these lines.
But i want to make this method more robust, maybe it was call by another function in future, and the caller may not know our assumptions.
The `schema` in the context is undefined and may cause an exception.

Comment (praiskup) [reply backend/copr_backend/msgbus.py:43]: This just feels weird, maybe the position in code, I'm not sure.

If things work for you just like that, can we just (and resolve the rest later?):
```
ry:
    from copr_messaging import schema
except ImportError:
    # copr_messaging is optional
    schema = None
```

**Comment by jamacku** [comment]
@praiskup It's GitHub's "new" feature, but it is messy. It requires you to upload SARIF on the `push` event as well.

I was dealing with this issue in:
- https://github.com/redhat-plumbers-in-action/differential-shellcheck/issues/215

**Review by praiskup — APPROVED** [review]

**Pull request merged**